pkg/workloadmeta/collectors/internal/process/process_test.go

```diff
@@ -0,0 +1,188 @@
+
+	return stream
+}
+
+func setUpCollectorTest(t *testing.T) *collectorTest {
+	t.Helper()
+
+	cfg := config.Mock(t)
+	port, err := testutil.FindTCPPort()
+	require.NoError(t, err)
+	cfg.Set("process_config.language_detection.grpc_port", port)
+	cfg.Set("workloadmeta.remote_process_collector.enabled", true)
+
+	wlmExtractor := workloadmetaExtractor.NewWorkloadMetaExtractor(cfg)
+	grpcServer := workloadmetaExtractor.NewGRPCServer(cfg, wlmExtractor)
+
+	mockProcessData, probe := checks.NewProcessDataWithMockProbe(t)
+	mockProcessData.Register(wlmExtractor)
+
+	oldCollector := c
+	t.Cleanup(func() {
+		c = oldCollector
+	})
+
+	store := workloadmeta.NewMockStore()
+
+	mockClock := clock.NewMock()
+
+	c := &collector{
+		ddConfig:        cfg,
+		processData:     mockProcessData,
+		wlmExtractor:    wlmExtractor,
+		grpcServer:      grpcServer,
+		pidToCid:        make(map[int]string),
+		collectionClock: mockClock,
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	require.NoError(t, c.Start(ctx, store))
+	t.Cleanup(cancel)
+
+	return &collectorTest{
+		collector: c,
+		probe:     probe,
+		clock:     mockClock,
+		store:     store,
+		stream:    acquireStream(t, port),
+	}
+}
+
+func (c *collectorTest) setupProcs() {
+	c.probe.On("ProcessesByPID", mock.Anything, mock.Anything).Return(map[int32]*procutil.Process{
+		1: {
+			Pid:     1,
+			Cmdline: []string{"some-awesome-game", "--with-rgb", "--give-me-more-fps"},
+			Stats:   &procutil.Stats{CreateTime: 1},
+		},
+	}, nil).Maybe()
+}
+
+func (c *collectorTest) waitForContainerUpdate(t *testing.T, cont *workloadmeta.Container) {
+	t.Helper()
+
+	c.store.SetEntity(cont)
+	require.EventuallyWithT(t, func(collect *assert.CollectT) {
+		assert.Contains(t, c.collector.pidToCid, cont.PID)
+	}, 5*time.Second, 1*time.Second)
+}
+
+func TestProcessCollector(t *testing.T) {
+	c := setUpCollectorTest(t)
+	c.setupProcs()
+
+	// Fast-forward through sync message
+	resp, err := c.stream.Recv()
+	require.NoError(t, err)
+
+	c.clock.Add(collectionInterval)
+	resp, err = c.stream.Recv()
+	assert.NoError(t, err)
+
+	require.Len(t, resp.SetEvents, 1)
+	evt := resp.SetEvents[0]
+	assert.EqualValues(t, 1, evt.Pid)
+	assert.EqualValues(t, 1, evt.CreationTime)
+
+	// Now test that this process updates with container id when the store is changed
+	c.waitForContainerUpdate(t, &workloadmeta.Container{
+		EntityID: workloadmeta.EntityID{
+			Kind: workloadmeta.KindContainer,
+			ID:   testCid,
+		},
+		PID: 1,
+	})
+
+	c.clock.Add(collectionInterval)
+	resp, err = c.stream.Recv()
+	assert.NoError(t, err)
+
+	require.Len(t, resp.SetEvents, 1)
+	evt = resp.SetEvents[0]
+	assert.EqualValues(t, 1, evt.Pid)
+	assert.EqualValues(t, 1, evt.CreationTime)
+	assert.Equal(t, testCid, evt.ContainerId)
+}
+
+// Assert that the collector is only Enabled if the process check is disabled and
+// the remote process collector is Enabled.
+func TestEnabled(t *testing.T) {
+	t.Run("process check Enabled", func(t *testing.T) {
```

Comment:
This test can be rewritten as a table test to avoid having to duplicate assertions

Reply:
Done!